wcfsetup/install/files/lib/system/user/notification/UserNotificationHandler.class.php

```diff
@@ -181,6 +181,10 @@ public function fireEvent($eventName, $objectType, IUserNotificationObject $noti
 					]);
 				}
 				WCF::getDB()->commitTransaction();
+
+				$parameters['notificationIDs'] = $notificationIDs;
+				EventHandler::getInstance()->fireAction($this, 'updateStackableNotifications', $parameters);
```

Comment:
You have the `fireEvent` event right above. Why is there another event required? You have all informations which are required to read the `$notificationID` by yourself.

Reply:
Yes, but that's causing duplications (or more) of the queries and php-logic.